testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/remote/EJBOverHTTPTestCase.java

```diff
@@ -0,0 +1,619 @@
+     * The key behaviour to validate: stickiness of EJB sessions to the nodes that own them.
+     * Proxies are obtained via JNDI/HTTP and invocations are made using EJB/HTTP.
+     */
+    public void testSFSBWithoutFailover(ExceptionSupplier<EJBDirectory, Exception> directoryProvider) throws Exception {
+
+        log.infof(MODULE_NAME+ " : testSFSBWithoutFailover: starting test");
+        try (EJBDirectory directory = directoryProvider.get()) {
+            // this single statement creates a session on the server
+            Incrementor sfsb = directory.lookupStateful(StatefulIncrementorBean.class, Incrementor.class);
+            logAffinityForBean(sfsb, "testSFSBWithoutFailover");
+
+            Result<Integer> result = sfsb.increment();
+            log.infof("Called SFSBWithoutFailover: SFSB value = %s, backend node = %s", result.getValue().intValue(), result.getNode());
+
+            int count = 1;
+            for (int i = 0; i < COUNT; ++i) {
+                result = sfsb.increment();
+                log.infof("Called SFSBWithoutFailover: SFSB value = %s, backend node = %s", result.getValue().intValue(), result.getNode());
+            }
+        }
+    }
+
+    /*
+     * A test which checks SFSB and SLSB behaviour of EJB client with EJB/HTTP in the absence of failover.
+     * The key behaviour to validate: stickiness of EJB SFSB sessions to the nodes that own them, while at the
+     * same time allowing SLSB EJBs to balance across nodes.
+     * Proxies are obtained via JNDI/HTTP and invocations are made using EJB/HTTP.
+     */
+    public void testSFSBAndSLSBWithoutFailover(ExceptionSupplier<EJBDirectory, Exception> directoryProvider) throws Exception {
+
+        log.infof(MODULE_NAME+ " : testSLSBWAndSLSBithoutFailover: starting test");
+        try (EJBDirectory directory = directoryProvider.get()) {
+            // this single statement creates a session on the server
+            Incrementor sfsb = directory.lookupStateful(StatefulIncrementorBean.class, Incrementor.class);
+            Incrementor slsb = directory.lookupStateless(StatelessIncrementorBean.class, Incrementor.class);
+
+            Result<Integer> sfsbResult = sfsb.increment();
+            log.infof("Called SFSBAndSLSBWithoutFailover: SFSB value = %s, backend node = %s", sfsbResult.getValue().intValue(), sfsbResult.getNode());
+
+            Result<Integer> slsbResult = slsb.increment();
+            log.infof("Called SFSBAndSLSBWithoutFailover: SLSB backend node = %s", slsbResult.getNode());
+
+            int count = 1;
+            for (int i = 0; i < COUNT; ++i) {
+                sfsbResult = sfsb.increment();
+                log.infof("Called SFSBAndSLSBWithoutFailover: SFSB value = %s, backend node = %s", sfsbResult.getValue().intValue(), sfsbResult.getNode());
+
+                slsbResult = slsb.increment();
+                log.infof("Called SFSBAndSLSBWithoutFailover: SLSB backend node = %s", slsbResult.getNode());
```

Comment:
@rachmatowicz perhaps this test might also cover what's tested in `testSLSBWithoutFailover` and `testSFSBWithoutFailover` and we can remove them.... or, maybe, I am missing some detail which differentiates the three of them?